provisioning_client/src/prov_transport_mqtt_common.c

```diff
@@ -319,7 +319,7 @@ static void mqtt_notification_callback(MQTT_MESSAGE_HANDLE handle, void* user_ct
             size_t status_pos = strlen(MQTT_TOPIC_STATUS_PREFIX);
             if (memcmp(MQTT_TOPIC_STATUS_PREFIX, topic_resp, status_pos) == 0)
             {
-                // If the status code is > 429 then this is a transient error
+                // If the status code is >= 429 then this is a transient error
                 long status_code = atol(topic_resp + status_pos);
                 if (status_code >= PROV_STATUS_CODE_TRANSIENT_ERROR)
```

Comment:
I don't think this is true.

From embedded SDK: (https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/inc/azure/iot/az_iot_common.h) 

AZ_NODISCARD AZ_INLINE bool az_iot_status_retriable(az_iot_status status)
{
  return ((status == AZ_IOT_STATUS_THROTTLED) || (status == AZ_IOT_STATUS_SERVER_ERROR));
}

The other errors appear to be non-retriable.

Reply:
Notice that this code keeps trying to reconnect no matter what. 
If we go for fixing the entire code to distinguish between retryable and non-retryable errors, there will be a lot more changes that this. 
This is fixing a very specific issue.

Reply:
But if we take the priorization away, I do agree with you.